src/engine/ItemmarkManager.ts

```diff
@@ -0,0 +1,105 @@
+import type { IMediaContainer } from './providers/MediaPlugin';
+import { type StorageController, Store } from './StorageController';
+
+export class ItemmarkManager {
+
+    private readonly Items:Map<string,ItemMark[]>=new Map();
+    constructor(private readonly storage: StorageController) {
+        this.Load();
+    }
+
+    private async Load() {
+        const mediamarks = await this.storage.LoadPersistent<ItemmarkSerialized[]>(Store.Itemmarks);
+        mediamarks.forEach(itemmark => {
+            this.Items.set(`${itemmark.Media.ProviderID} :: ${itemmark.Media.EntryID}`,itemmark.Marks);
+        });
+    }
+
+    private Serialize(media: IMediaContainer, marks: ItemMark[]): ItemmarkSerialized {
+        return {
+            Media: {
+                ProviderID: media.Parent.Parent.Identifier,
+                EntryID: media.Parent.Identifier
+            },
+            Marks: marks,
+        };
+    }
+
+    /* Retrieves the marks of a media */
+    public Find(entry: IMediaContainer): ItemMark[] {
```

Comment:
Is this public by intention? Seems more like it was meant to be private

Reply:
Yes, the UI fetches the marks of a media, and caches it as the UI will call the getMarks a lot
Used here: https://github.com/manga-download/haruneko/pull/51/files#diff-cc54b4e21d800248a19197cf1c4d48632d290d108ec4f403a3fa137e89679ac7R72

Reply:
Consider using a more expressive method name

Reply:
If performance is a concern, avoid caliing the `getMark` method at all from UI (it calculates hashes for each call).
Consider providing a map instead and making `getMark` private

e.g.
```
// Create a map that holds the mark for each chapter of the given manga
Find(container) {
    const cache = new Map<IMediaContainer, MarkType>();
    const marks = this.Items.get(this.StorageKey(entry));
    for(const entry of container.Entries) {
        cache.set(entry, this.getMark(entry));
    }
    return cache;
}
```

Reply:
one of the main "issue" with svelte is that it uses assign statement (=) to update a component, but i'll try to find if i can get an alternative way to forces it to detect a change on a map element (I read something about keyed elements in svelte)